nixos/modules/programs/wayland/uwsm.nix

```diff
@@ -0,0 +1,129 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.programs.uwsm;
+
+  # Helper function to create desktop entry files for UWSM-managed compositors
+  mk_uwsm_desktop_entry =
+    opts:
+    (pkgs.writeTextFile {
+      name = "${opts.name}-uwsm";
+      text = ''
+        [Desktop Entry]
+        Name=${opts.prettyName} (UWSM)
+        Comment=${opts.comment}
+        Exec=${lib.getExe cfg.package} start -S -F -N "${opts.prettyName}" -- "${opts.binPath}"
```

Comment:
This doesn't work for me with `sddm`.
After a few quick tests, it seems to be caused by the escaping of `"`.
But I think we can also remove `-N "${opts.prettyName}"` since `uwsm` will read the name from the `Name` entry either way (if I understand the help output correctly) and I do not see a reason why it should only be `prettyName` and not `prettyname (UWSM)`.
For the binPath, I am not sure if the `path` type already escapes the path. :thinking:

Reply:
Try now. It did work for me on GDM though.

Reply:
> But I think we can also remove -N "${opts.prettyName}" since uwsm will read the name from the Name entry either way (if I understand the help output correctly) and I do not see a reason why it should only be prettyName and not prettyname (UWSM).

UWSM will not have access to desktop entry if its main argument (`"${opts.binPath}"`) is compositor's binary. In this case entry's relevant fields should be provided via `-N`, `-D`. If the main argument is another entry, then it will be parsed.

Reply:
:facepalm: Yeah, true. Sorry for the confusion.
Without the `"` it does work for me, but if we want to add the pretty name, we need `"` or `'` due to the space. I will have to take a closer look at the sddm related bits to understand what is happening.

Reply:
Desktop entries [should use double quotes](https://specifications.freedesktop.org/desktop-entry-spec/latest/exec-variables.html)

Reply:
`-D` is not necessary, assuming DM will set `$XDG_CURRENT_DESKTOP` properly. And unless you want to force names with `-e`

Reply:
> Desktop entries [should use double quotes](https://specifications.freedesktop.org/desktop-entry-spec/latest/exec-variables.html)

I agree. But, there seems to be a bug in the way SDDM handles it.
If I add quotes to the `Exec` entry, it stops working (currently only tested in combination with `uwsm`).
It might have something to do with this line:

https://github.com/sddm/sddm/blob/4ec29a8bba033d475f197693fac6cb0c383a1da2/data/scripts/wayland-session#L49

I first thought that simply wrapping it in `"$@"` would fix it, but that seems to cause a different issue.
I am guessing that with the quotes, it tries to interpret the entire command as a single command, and I get a 127 exit code.

I will take a closer look over the weekend.

Reply:
I was also having an issue with GDM.

It works if I type something like in the terminal
```shell
/nix/store/zn64gjhb383imvwhh73vdzqvxzvxw01w-uwsm-0.17.4/bin/uwsm start -S -N "Hyprland" -C "Hyprland compositor managed by UWSM" -- "/run/current-system/sw/bin/Hyprland"
```

but selecting the session from GDM and entering my password makes it unable to login.

I guess there could be a character limit in GDM? CC @jtojnar 

Removing -N and -C parameters makes it work on GDM.

```shell
/nix/store/zn64gjhb383imvwhh73vdzqvxzvxw01w-uwsm-0.17.4/bin/uwsm start -S -- "/run/current-system/sw/bin/Hyprland"
```

For SDDM issue: CC @K900, KDE maintainer

Reply:
> I first thought that simply wrapping it in "$@" would fix it, but that seems to cause a different issue.
I am guessing that with the quotes around $@, it tries to interpret the entire command as a single command, and I get a 127 exit code.

`"$@"` is The correct way to pass argument array as is. If it is lumped in one arg, it happened before this point.